lib/Mojo/WebSocketProxy/Backend/ConsumerGroups.pm

```diff
@@ -0,0 +1,345 @@
+Stucture of the hash should be like:
+
+=over 4
+
+=item * C<key> - request id, which we'll get from redis after successful adding request to the stream
+
+=item * C<value> - future object, which will be done in case of getting response, of cancelled in case of timeout
+
+=back
+
+=cut
+
+sub pending_requests {
+    shift->{pending_requests} //= {};
+}
+
+=head2 redis
+
+=cut
+
+sub redis {
+    my $self = shift;
+    return $self->{redis} //= Mojo::Redis2->new(url => $self->{redis_uri});
+}
+
+=head2 timeout
+
+=cut
+
+sub timeout {
+    return shift->{timeout} //= RESPONSE_TIMEOUT;
+}
+
+=head2 whoami
+
+Return unique ID of Redis which will be used by backend server to send response.
+Id is persistent for the object.
+
+=cut
+
+sub whoami {
+    my $self = shift;
+
+    return $self->{whoami} if $self->{whoami};
+
+    Math::Random::Secure::srand() if Math::Random::Secure->can('srand');
+    $self->{whoami} = Data::UUID->new->create_str();
+
+    return $self->{whoami};
+}
+
+=head2 call_rpc
+
+Makes a remote call to a process  returning the result to the client in JSON format.
+Before, After and error actions can be specified using call backs.
+It takes the following arguments
+
+=over 4
+
+=item * C<$c>  : L<Mojolicious::Controller>
+
+=item * C<$req_storage> A hashref of attributes stored with the request.  This routine uses some of the following named arguments:
+
+=over 4
+
+=item * C<method> The name of the method at the remote end.
+
+=item * C<msg_type> a name for this method; if not supplied C<method> is used.
+
+=item * C<call_params> a hashref of arguments on top of C<req_storage> to send to remote method. This will be suplemented with C<< $req_storage->{args} >>
+added as an C<args> key and be merged with C<< $req_storage->{stash_params} >> with stash_params overwriting any matching 
+keys in C<call_params>. 
+
+=item * C<rpc_response_callback>  If supplied this will be run with args: C<< Mojolicious::Controller >> instance, the rpc_response and C<< $req_storage >>.
+B<Note:> if C<< rpc_response_callback >> is supplied the success and error callbacks are not used. 
+
+=item * C<before_get_rpc_response>  arrayref of subroutines to be run before the remote response is received, with args: C<< $c >> and C<< req_storage >>
+
+=item * C<after_get_rpc_response> arrayref of subroutines to be run after the remote response is received; called with args: C<< $c >> and C<< req_storage >>
+called only when there is an response from the remote call.
+
+=item * C<before_call> arrayref of subroutines called before the request to the remote service is made.
+
+=item * C<rpc_failure_cb> a subroutine reference to call if the remote call fails. Called with C<< Mojolicious::Controller >>, the rpc_response and C<< $req_storage >>
+
+=back
+
+=back
+
+Returns undef.
+
+=cut
+
+sub call_rpc {
+    my ($self, $c, $req_storage) = @_;
+
+    # make sure that we are already waiting for messages
```

Comment:
What does this refer to? There's no code next to these comments and it's not clear how it applies to the `->call_rpc` method itself?

Reply:
Fixed. i removed outdated comment